### ISSUE TITLE
MediaPlayerStateChanged Buffering

### ISSUE THREAD

**Issue by jaruba** [RSATom/WebChimera#113]
``MediaPlayerStateChanged`` does not trigger for the buffering state. See this demo:

http://editor.webchimera.org/c/1429781968

**Comment by RSATom** [comment]
hm... strange, it should (I've checked the sources). Ok, will look close a little later. Thanks!

**Comment by jaruba** [comment]
then maybe it's something with ``.state``? could it be it doesn't change to 3 (buffering)? or maybe ``.state`` is changed to buffering after the buffering event is triggered?

**Comment by RSATom** [comment]
I think `.state` could skip some state if change was happen very quickly, since state event not tied strong with .state value. And there could be some delay in .state update. Events in this case are more reliable.

**Comment by RSATom** [comment]
The reason is I not emit `MediaPlayerStateChanged` when .state value was changed (I have no control over it), but when some related event arrive: https://github.com/RSATom/QmlVlc/blob/master/QmlVlcPlayerProxy.cpp#L50

So if you get `MediaPlayerBuffering`event, after that you will get `MediaPlayerStateChanged` since one trigger another.

**Comment by jaruba** [comment]
Does ``MediaPlayerStateChanged`` also pass the new event to the callback function?

**Comment by RSATom** [comment]
Don't sure understand what you mean.

**Comment by jaruba** [comment]
if something like ``MediaPlayerStateChanged( state )`` is possible

**Comment by RSATom** [comment]
no, since I have only async access to state. I.e. it will have the same problem.

**Comment by RSATom** [comment]
but stop, maybe you are right, I can deduce current state from original event. Good Idea.

**Comment by jaruba** [comment]
k, understood. Then I don't think there is any real solution to it. :)

**Comment by jaruba** [comment]
There is one more problem with ``MediaPlayerStateChanged`` that I've found. I guess it should trigger only when the state has changed (ie: is different from the previous state), but I added a ``console.log()`` to the event and it's triggering a lot more times for the same exact states in a row.

**Comment by RSATom** [comment]
Yes, I didn't store previous state value
